### PR TITLE
TC-4-Logout-User

### PR DESCRIPTION
**Pull request by abdelazizaliGS**
I have added a test automation script to logout user

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
     <properties>
         <!-- Make sure you point to the latest release version: https://github.com/ShaftHQ/SHAFT_ENGINE/releases -->
-        <shaft_engine.version>8.1.20240115</shaft_engine.version>
+        <shaft_engine.version>8.1.20240210</shaft_engine.version>
```
Comment (MahmoudElSharkawyGS) [inline pom.xml:13]: Why did you change the shaft version?
Also it's not part of the changes scope.

File: src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    //Variables
+    private SHAFT.GUI.WebDriver driver;
+    private final String url = System.getProperty("baseUrl");
+
+
+
+    //Locators
+
+
+    private final By Logout_button = By.xpath("//header//ul//li[4]/a");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java:19]: This is not a relative path, please make sure to enhance it with relative one.

File: src/test/java/com/gizasystems/automationexercise/tests/logoutUser/logOutUser.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    private SHAFT.TestData.JSON testData;
+
+    private String url = System.getProperty("url");
+    @Test
+    public void SuccessfulLogOut(){
+
+
+
+        new com.gizasystems.automationexercise.pages.LogoutUser(driver).logOut();
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/logoutUser/logOutUser.java:35]: Why do you have to write the full path on the class here; Please import it.

File: src/test/java/com/gizasystems/automationexercise/tests/logoutUser/logOutUser.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @BeforeMethod
+    public void beforeMethod() {
+        driver = new SHAFT.GUI.WebDriver();
+        new HomePage(driver)
+                .navigate()
+                .validateOnVisibilityOfHomePage();
+        new NavigationBar(driver).clickOnSignupLoginLink();
+        new SignupLoginPage(driver).registeredUserLogin("abdelazizaliabdelaziz@gmail.com" ,"karimAhmed");
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/logoutUser/logOutUser.java:52]: Please make sure that all the test data are in a json test data file.

Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/logoutUser/logOutUser.java:52]: Where's the registration step? I believe that it's a prerequisite for your case. (You can use the API method to register and then login from UI as you did already)

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
- Please make sure to execute the pipeline and share the link as agreed
- Please reformat your code
- Please delete any excessive empty lines of code as per conventions 
- Please revert any changes related to the properties default folder

**Comment by abdelazizaliGS** [comment]
image (1)

File: src/test/resources/testDataFiles/LoginUser.json [hunk 5]
```diff
@@ -10,5 +10,7 @@
     "AccountCreated": "ACCOUNT CREATED!",
     "AccountDeleted": "ACCOUNT DELETED!",
     "Login": "Login to your account"
-  }
+  },
+  "UserEMail" : "abdelazizaliabdelaziz@gmail.com",
+  "UserPassWD" : "karimAhmed"
```
Comment (MahmoudElSharkawyGS) [inline src/test/resources/testDataFiles/LoginUser.json:15]: - We agreed not to use our names in any of the test data.
- Also, please make sure to create a new test data json file for your test class, and not to use other test data json file for other test class

File: src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.shaft.driver.SHAFT;
+import io.qameta.allure.Step;
+import org.openqa.selenium.By;
+
+public class LogoutUser {
+
+    //Variables
+    private SHAFT.GUI.WebDriver driver;
+    private final String url = System.getProperty("baseUrl");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java:11]: Where is the link to the page itself here?

File: src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java:1]: Do we have a page called Logout? or it's just a button on the navigation bar?

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]

**Comment by MahmoudElSharkawyGS** [comment]
Also, please make sure to share the pipeline execution link, not a screenshot.

File: src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java
Comment (abdelazizaliGS) [reply src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java:1]: just a button on the navigation bar

File: src/test/resources/testDataFiles/LoginUser.json (hunk 5 above)
Comment (abdelazizaliGS) [reply src/test/resources/testDataFiles/LoginUser.json:15]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java (hunk 6 above)
Comment (abdelazizaliGS) [reply src/main/java/com/gizasystems/automationexercise/pages/LogoutUser.java:11]: It is not a page 
it is just a button in the navigation bar

File: src/test/resources/testDataFiles/LogoutTestData.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "UserEMail" : "lenesah401@mcuma.com",
```
Comment (MahmoudElSharkawyGS) [inline src/test/resources/testDataFiles/LogoutTestData.json:2]: Can you add something related to automation like auto or bot like we are doing on other cases?

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
PLEASE EXECUTE THE PIPELINE :D

**Review by MahmoudElSharkawyGS — APPROVED** [review]
